### PR TITLE
Adding initial docs for filtering events

### PR DESCRIPTION
**Pull request by eduardonunesp**

### CODE REVIEW

File: docs/web3js-event-filters.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+```
+
+## Filtering indexed values
+
+Another great feature is the filtering by `indexed` values, which can be used to filter specific events on contracts, for instance 
+on a contract that a specific event is only triggered if a certain `indexed` value happen.
+
+For the following contract:
+
+```js
```
Comment (enlight) [inline docs/web3js-event-filters.md]: This isn't JS code, do we have a Solidity syntax highlighter?

File: docs/web3js-event-filters.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+// Watch filter will return the hash of the latest block continuously 
+filter.watch(function (error, result) {
+  if (error) {
+    console.error(error)
+  } else {
+    console.log('Block hash', result)
+```
+
+## Filtering indexed values
```
Comment (enlight) [inline docs/web3js-event-filters.md]: ->`Filtering by indexed values`

File: docs/web3js-event-filters.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+Another great feature is the filtering by `indexed` values, which can be used to filter specific events on contracts, for instance 
+on a contract that a specific event is only triggered if a certain `indexed` value happen.
```
Comment (enlight) [inline docs/web3js-event-filters.md]:
```
Another great feature is filtering by `indexed` values, which can be used to trigger event handlers when a specific `indexed` value is emitted.
```

File: docs/web3js-event-filters.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  event NewValueSet(uint _value);
+
+  function set(uint _value) public {
+    value = _value;
+    emit NewValueSet(value);
+  }
+}
+```
+
+Is possible to use the following code to subscribe and filter `indexed` values on events.
```
Comment (enlight) [inline docs/web3js-event-filters.md]:
```
It's possible to setup an event handler for the `NewValueSet` event that's only triggered when the `value` emitted is `10`, and it won't be triggered if the contract emits any other value.
```

File: docs/web3js-event-filters.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+pragma solidity ^0.4.22;
+
+contract SimpleStore {
+  uint value;
+
+  constructor() {
+      value = 10;
+  }
+
+  event NewValueSet(uint _value);
```
Comment (enlight) [inline docs/web3js-event-filters.md]: Don't you need to mark the value as `indexed`?

File: docs/web3js-event-filters.md (hunk 1 above)
Comment (eduardonunesp) [reply docs/web3js-event-filters.md]: I don't know

File: docs/web3js-event-filters.md (hunk 5 above)
Comment (eduardonunesp) [reply docs/web3js-event-filters.md]: Yep ! holy ship copied the wrong example !

**Pull request merged**